did the Kata exercise. Learned a lot. As always, more questions than ans

scotchwsplenda opened the pull request.

charlesrobison commented on students/scotchwsplenda/Lesson_4/Rabbit_Hole.py:
```diff
@@ -0,0 +1,72 @@
+
+with open('Alice_In_Wonder.txt', 'r', encoding='utf8') as all_the_words:
+    # https://www.geeksforgeeks.org/python-removing-unwanted-characters-from-string/
+    bad_punct=['.','*','(',')','!',',','?','--','[',']']
```
Missing whitespace around assignment operator, as well as after commas:

bad_punct = ['.', '*', etc.

charlesrobison commented on students/scotchwsplenda/Lesson_4/Rabbit_Hole.py:
```diff
@@ -0,0 +1,72 @@
+    # didn't work after the replace
+    split_it = all_the_words.split()
+
+dicty={}
```
Whitespacing: dicty = {}
You typically want to give meaningful names too.

charlesrobison commented on students/scotchwsplenda/Lesson_4/Rabbit_Hole.py:
```diff
@@ -0,0 +1,72 @@
+
+dicty={}
+for i in range(len(split_it)-2):
+    pair=split_it[i:i+2]
```
Whitespace around this equals sign, as well as the rest below

scotchwsplenda commented:
What does a linter do?

On Sat, Feb 8, 2020 at 9:28 PM Charles Robison <notifications@github.com>
wrote:

> *@charlesrobison* approved this pull request.
>
> Code works well - just some whitespace formatting issues. @scotchwsplenda
> <https://github.com/scotchwsplenda> do you have a linter installed on
> your text editor? If not, highly recommended.
>
> Ready to merge
> ------------------------------
>
> In students/scotchwsplenda/Lesson_4/Rabbit_Hole.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/722#discussion_r376756240>
> :
>
> > @@ -0,0 +1,72 @@
> +import random
> +
> +with open('Alice_In_Wonder.txt', 'r', encoding='utf8') as all_the_words:
> +    # https://www.geeksforgeeks.org/python-removing-unwanted-characters-from-string/
> +    bad_punct=['.','*','(',')','!',',','?','--','[',']']
>
> Missing whitespace around assignment operator, as well as after commas:
>
> bad_punct = ['.', '*', etc.
> ------------------------------
>
> In students/scotchwsplenda/Lesson_4/Rabbit_Hole.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/722#discussion_r376756276>
> :
>
> > @@ -0,0 +1,72 @@
> +import random
> +
> +with open('Alice_In_Wonder.txt', 'r', encoding='utf8') as all_the_words:
> +    # https://www.geeksforgeeks.org/python-removing-unwanted-characters-from-string/
> +    bad_punct=['.','*','(',')','!',',','?','--','[',']']
> +    all_the_words = all_the_words.read()
> +    for i in bad_punct:
> +        all_the_words=all_the_words.replace(i,'')
> +    # https://stackoverflow.com/questions/12572362/get-a-string-after-a-specific-substring/12572391
> +    # didn't work after the replace
> +    split_it = all_the_words.split()
> +
> +dicty={}
>
> Whitespacing: dicty = {}
> You typically want to give meaningful names too.
> ------------------------------
>
> In students/scotchwsplenda/Lesson_4/Rabbit_Hole.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/722#discussion_r376756298>
> :
>
> > @@ -0,0 +1,72 @@
> +import random
> +
> +with open('Alice_In_Wonder.txt', 'r', encoding='utf8') as all_the_words:
> +    # https://www.geeksforgeeks.org/python-removing-unwanted-characters-from-string/
> +    bad_punct=['.','*','(',')','!',',','?','--','[',']']
> +    all_the_words = all_the_words.read()
> +    for i in bad_punct:
> +        all_the_words=all_the_words.replace(i,'')
> +    # https://stackoverflow.com/questions/12572362/get-a-string-after-a-specific-substring/12572391
> +    # didn't work after the replace
> +    split_it = all_the_words.split()
> +
> +dicty={}
> +for i in range(len(split_it)-2):
> +    pair=split_it[i:i+2]
>
> Whitespace around this equals sign, as well as the rest below
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/722?email_source=notifications&email_token=AM2M4S4XJVYY3HXSZPNDYALRB6H7JA5CNFSM4KQBZXS2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCUY4U2Q#pullrequestreview-355584618>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM2M4S7CQMZBJ6UMVC2WB33RB6H7JANCNFSM4KQBZXSQ>
> .
>
-- 
*Oliver B. Lockwood*

charlesrobison commented:
@scotchwsplenda - it points out all of the non-Python PEP8 code issues.  Check out this for example: https://dbader.org/blog/python-code-linting
